projects/common/src/navigation/navigation.service.ts

```diff
@@ -80,11 +82,16 @@ export class NavigationService {
 
     if (params.navType === NavigationParamsType.External) {
       // External
+      const queryParamString = getQueryParamStringFromObject(params.queryParams ?? {});
```

Comment:
It might be better to let angular do any url building for us (such as url tree). This would break if we passed it a a url that contained query params, since it wouldn't merge. Say `http://website.com?param1=value` and a second param via query params `{ param2=value }`, we'd get `http://website.com?param1=value?param2=value` instead of `http://website.com?param1=value&param2=value`.

I think you can combine `router.parseUrl`, `router.serializeUrl` and maybe `router.createUrlTree` to get all the right behavior here.